### ISSUE TITLE
[Bug] Using PublicClientApplicationBuilder and WithAuthority(string cloudInstanceUri, string tenant, bool validateAuthority) throws exception when acquiring token

### ISSUE THREAD

**Issue by Crazyconv** [AzureAD/microsoft-authentication-library-for-dotnet#1270]
**Which Version of MSAL are you using ?**
MSAL 4.1.0

**Platform**
net45

**What authentication flow has the issue?**
* Desktop / Mobile
    * Username Password

**Is this a new or existing app?**
This is a new app

**Repro**

```csharp
string tenantId = "<tenantId";
string username = "<username>";
string password = "<password>";
string clientId = "<clientId>";

string[] permissions = new[] {"<permission>"};
IPublicClientApplication app = PublicClientApplicationBuilder.Create(clientId).WithAuthority("https://login.microsoftonline.com", tenantId).Build();
AuthenticationResult result = await app.AcquireTokenByUsernamePassword(permissions, username, this.ToSecureString(password)).ExecuteAsync();
```

**Expected behavior**
An AuthenticationResult result is returned.

**Actual behavior**
Exception is thrown.
Microsoft.Identity.Client.MsalServiceException: "AADSTS90002: Tenant 'v2.0' not found. This may happen if there are no active subscriptions for the tenant. Check with your subscription administrator.\r\nTrace ID: 024c3425-f6b7-4f00-a95b-8b0a14325400\r\nCorrelation ID: d2d97559-d4ab-4517-9a3f-bec02f562a50\r\nTimestamp: 2019-07-15 07:33:33Z"

**Possible Solution**
In [this line](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/e2bcfcbb41d529fdb34af7550ce122e70e9b4566/src/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs#L94), `string.Format(CultureInfo.InvariantCulture, "{0}/{1}/", cloudInstanceUri, tenant)` produces `https://login.microsoftonline.com//<tenantId>`, and so `CanonicalAuthority` of the `AuthorityInfo` object becomes `https://login.microsoftonline.com//`.

Updating the line to `string.Format(CultureInfo.InvariantCulture, "{0}{1}/", cloudInstanceUri, tenant)` should fix the issue. However I am not sure about other possible urls that can be passed into the method so not sure whether it will break other user cases.

**Comment by trwalke** [comment]
Hi @Crazyconv I believe I have a working solution for you. Do you have an email where I can send a preview nuget for you to verify the fix?

**Comment by Crazyconv** [comment]
@trwalke Thanks for the quick response. My email is conanconv@gmail.com.

**Comment by jennyf19** [comment]
Included in [4.3.0](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.3.0) release
